apps/astarte_appengine_api/lib/astarte_appengine_api/device/queries.ex

```
@@ -683,41 +696,28 @@ defmodule Astarte.AppEngine.API.Device.Queries do
-      DELETE aliases[:alias_tag]
-      FROM devices
-      WHERE device_id = :device_id
-      """
-
-      delete_alias_from_device_query =
-        DatabaseQuery.new()
-        |> DatabaseQuery.statement(delete_alias_from_device_statement)
-        |> DatabaseQuery.put(:alias_tag, alias_tag)
-        |> DatabaseQuery.put(:device_id, device_id)
-        |> DatabaseQuery.consistency(:each_quorum)
-        |> DatabaseQuery.convert()
-
-      delete_alias_from_names_statement = """
-      DELETE FROM names
-      WHERE object_name = :alias AND object_type = 1
-      """
-
-      delete_alias_from_names_query =
-        DatabaseQuery.new()
-        |> DatabaseQuery.statement(delete_alias_from_names_statement)
-        |> DatabaseQuery.put(:alias, alias_value)
-        |> DatabaseQuery.put(:device_id, device_id)
-        |> DatabaseQuery.consistency(:each_quorum)
-        |> DatabaseQuery.convert()
-
-      delete_batch =
-        CQEx.cql_query_batch(
-          consistency: :each_quorum,
-          mode: :logged,
-          queries: [delete_alias_from_device_query, delete_alias_from_names_query]
+      map_new_alias = MapSet.new([alias_tag])
+
+      query_delete_alias =
+        from ApiDevice,
+          prefix: ^keyspace,
+          where: [device_id: ^device_id],
+          update: [set: [aliases: fragment("aliases - ?", ^map_new_alias)]]
+
+      sql_query_delete_alias = Repo.to_sql(:update_all, query_delete_alias)
+
+      query_delete_in_name =
+        from(d in Name,
+          prefix: ^keyspace,
+          where: [object_name: ^alias_value, object_type: 1]
         )
 
-      with {:ok, _result} <- DatabaseQuery.call(client, delete_batch) do
+      sql_query_delete_in_name = Repo.to_sql(:delete_all, query_delete_in_name)
+
+      update_and_delete_batch =
+        %Exandra.Batch{queries: [sql_query_delete_alias, sql_query_delete_in_name]}
+
+      with :ok <- Exandra.execute_batch(Repo, update_and_delete_batch, consistency: :each_quorum) do
```

`Exandra.execute_batch` should return `:ok`, or `{:error, exception}`.
If we can "let it crash" like in other API services, we can avoid handling the error case here, so the with/else block can be replaced with:

```elixir
Exandra.execute_batch(Repo, update_and_delete_batch, consistency: :each_quorum)
```